src/Diagnostics/atmos_default.jl

```diff
@@ -194,65 +241,87 @@ function compute_diagnosticsums!(
+    avg_ht_sgs = ha.ρht_sgs / ha.ρ
 
     # state and functions of state
-    ds.u += MH * ũ
-    ds.v += MH * ṽ
-    ds.w += MH * w̃
-    ds.e_tot += MH * ẽ
-    ds.q_tot += MH * ha.ρq_tot / ha.ρ
-    ds.q_liq += MH * ha.q_liq
-    ds.thd += MH * ha.θ_dry
-    ds.thl += MH * ha.θ_liq_ice
-    ds.thv += MH * ha.θ_v
-    ds.e_int += MH * ha.e_int
-    ds.h_m += MH * ha.h_m
-    ds.h_t += MH * ha.h_t
-    ds.qt_sgs += MH * ha.qt_sgs
-    ds.ht_sgs += MH * ha.ht_sgs
-
-    # vertical fluxes
-    ds.vert_eddy_mass_flux += MH * (w - w̃) * (state.ρ - ha.ρ)
-    ds.vert_eddy_u_flux += MH * (w - w̃) * (u - ha.ρu / ha.ρ)
-    ds.vert_eddy_v_flux += MH * (w - w̃) * (v - ha.ρv / ha.ρ)
-    ds.vert_eddy_ql_flux += MH * (w - w̃) * (th.q_liq - ha.q_liq)
-    ds.vert_eddy_qv_flux += MH * (w - w̃) * (th.q_vap - ha.q_vap)
-    ds.vert_eddy_thd_flux += MH * (w - w̃) * (th.θ_dry - ha.θ_dry)
-    ds.vert_eddy_thv_flux += MH * (w - w̃) * (th.θ_v - ha.θ_v)
-    ds.vert_eddy_thl_flux += MH * (w - w̃) * (th.θ_liq_ice - ha.θ_liq_ice)
-    # TODO: temporary fix
+    ds.rho += MH * ha.ρ
+    ds.u += MH * avg_u
+    ds.v += MH * avg_v
+    ds.w += MH * avg_w
+    ds.et += MH * avg_e_tot
+    ds.ql += MH * avg_q_liq
+    ds.thd += MH * avg_θ_dry
+    ds.thl += MH * avg_θ_liq_ice
+    ds.thv += MH * avg_θ_v
+    ds.ei += MH * avg_e_int
+    ds.hm += MH * avg_h_moi
+    ds.ht += MH * avg_h_tot
+    ds.qt_sgs += MH * avg_qt_sgs
+    ds.ht_sgs += MH * avg_ht_sgs
+
+    # covariances and variances
+    ds.cov_w_rho += MH * (w - avg_w) * (state.ρ - ha.ρ)
+    ds.cov_w_u += MH * (w - avg_w) * (u - avg_u)
+    ds.cov_w_v += MH * (w - avg_w) * (v - avg_v)
+    ds.cov_w_ql += MH * (w - avg_w) * (q_liq - avg_q_liq)
+    ds.cov_w_qv += MH * (w - avg_w) * (q_vap - avg_q_vap)
+    ds.cov_w_thd += MH * (w - avg_w) * (θ_dry - avg_θ_dry)
+    ds.cov_w_thl += MH * (w - avg_w) * (θ_liq_ice - avg_θ_liq_ice)
+    ds.cov_w_thv += MH * (w - avg_w) * (θ_v - avg_θ_v)
+    ds.cov_w_ei += MH * (w - avg_w) * (e_int - avg_e_int)
+
+    ds.var_u += MH * (u - avg_u)^2
+    ds.var_v += MH * (v - avg_v)^2
+    ds.var_w += MH * (w - avg_w)^2
+    ds.var_ei += MH * (e_int - avg_e_int)^2
+    ds.var_thl += MH * (θ_liq_ice - avg_θ_liq_ice)^2
+
+    # TODO: specialize on moisture model
     if isa(atmos.moisture, EquilMoist)
-        ds.vert_eddy_qt_flux += MH * (w - w̃) * (q_tot - q̃_tot)
-        ds.vert_qt_flux += MH * w * q_tot
+        q_tot = state.moisture.ρq_tot / state.ρ
+        avg_q_tot = ha.ρq_tot / ha.ρ
+
+        ds.qt += MH * avg_q_tot
+        ds.cov_w_qt += MH * (w - avg_w) * (q_tot - avg_q_tot)
+        ds.w_qt += MH * w * q_tot
+        ds.cov_qt_thl += MH * (q_tot - avg_q_tot) * (θ_liq_ice - avg_θ_liq_ice)
+        ds.cov_qt_ei += MH * (q_tot - avg_q_tot) * (e_int - avg_e_int)
+        ds.var_qt += MH * (q_tot - avg_q_tot)^2
     end
 
-    # variances
-    ds.uvariance += MH * (u - ũ)^2
-    ds.vvariance += MH * (v - ṽ)^2
-    ds.wvariance += MH * (w - w̃)^2
-
     # skewness
```

Comment:
#skewness should be replaced with #w3 since they are not the same quantity. Skewness can be obtained from w3 and var_w so that it's not necessary to calculate it here.